### PR TITLE
system-variables: add documents for `error_count` and `max_connections`

### PR DESCRIPTION
**Pull request by bb7133**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

As stated in the title, add missing documents for 2 sys vars. I also would like to add some more descriptions for `ssl_cert`, `ssl_key` and `ssl_ca`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by dveeden — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -335,6 +341,16 @@ This variable is an alias for `last_insert_id`.
...
 - The maximum size of a packet for the MySQL protocol.
 
+### max_connections
+
+- Scope: GLOBAL
+- Persists to cluster: No
+- Type: Integer
+- Default value: `0`
+- Range: `[0, 100000]`
+- The maximum number of connections for a single TiDB instance.
```
Comment (TomShawn) [inline system-variables.md:351]:
```suggestion
- The maximum number of connections permitted for a single TiDB instance.
```

File: system-variables.md [hunk 2]
```diff
@@ -501,19 +517,19 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 
 - Scope: NONE
 - Default value: ""
-- The location of the certificate authority file (if there is one).
+- The location of the certificate authority file (if there is one), the value of this variable is defined by [`ssl-ca` configuration](https://docs.pingcap.com/tidb/dev/tidb-configuration-file#ssl-ca).
```
Comment (TomShawn) [inline system-variables.md:520]:
```suggestion
- The location of the certificate authority file (if there is one), the value of this variable is defined by the TiDB configuration item [`ssl-ca`](/tidb-configuration-file.md#ssl-ca).
```

File: system-variables.md [hunk 3]
```diff
@@ -501,19 +517,19 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
...
 
 ### ssl_cert
 
 - Scope: NONE
 - Default value: ""
-- The location of the certificate file (if there is a file) that is used for SSL/TLS connections.
+- The location of the certificate file (if there is a file) that is used for SSL/TLS connections, the value of this variable is defined by [`ssl-cert` configuration](https://docs.pingcap.com/tidb/dev/tidb-configuration-file#ssl-cert).
```
Comment (TomShawn) [inline system-variables.md:526]:
```suggestion
- The location of the certificate file (if there is a file) that is used for SSL/TLS connections, the value of this variable is defined by the TiDB configuration item [`ssl-cert`](/tidb-configuration-file.md#ssl-cert).
```

File: system-variables.md [hunk 4]
```diff
@@ -501,19 +517,19 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
...
 
 ### ssl_key
 
 - Scope: NONE
 - Default value: ""
-- The location of the private key file (if there is one) that is used for SSL/TLS connections.
+- The location of the private key file (if there is one) that is used for SSL/TLS connections, the value of this variable is defined by [`ssl-key` configuration](https://docs.pingcap.com/tidb/dev/tidb-configuration-file#ssl-cert).
```
Comment (TomShawn) [inline system-variables.md:532]:
```suggestion
- The location of the private key file (if there is one) that is used for SSL/TLS connections, the value of this variable is defined by [`ssl-key`](/tidb-configuration-file.md#ssl-cert).
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**